src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConverters.java

```diff
@@ -535,6 +527,86 @@ public static RedisURI sentinelConfigurationToRedisURI(RedisSentinelConfiguratio
 		return builder.build();
 	}
 
+	/**
+	 * Converts a {@link RedisURI} to its corresponding {@link RedisSentinelConfiguration}.
+	 *
+	 * @param redisURI the uri containing the Redis Sentinel connection info
+	 * @return a {@link RedisSentinelConfiguration} representing the Redis Sentinel information in the Redis URI.
+	 * @since 2.6
+	 */
+	static RedisSentinelConfiguration redisUriToSentinelConfiguration(RedisURI redisURI) {
+
+		Assert.notNull(redisURI, "RedisURI is required");
```

Comment:
These `notNull` asserts are a bit redundant currently as the only place that calls this current is `LettuceConnectionFactory.createRedisConfiguration` which itself does this null check. The only reason I left it in here is in case it gets called elsewhere (package protected).

Reply:
Guarding `public` methods only is sufficient. I can take care of this during the merge.